### DISCUSSION TITLE
Conduction in Porous Flow from lava lamp example

### DISCUSSION THREAD

**Discussion by aikubo** [idaholab/moose#26122] [Q&A Modules: Porous Flow] [answered]
Hello, I'm trying to set up a simple convection problem using the PorousFlow module. I've modified the example problem ['/moose_projects/moose/modules/porous_flow/examples/lava_lamp/1phase_convection.i'](https://github.com/idaholab/moose/blob/next/modules/porous_flow/examples/lava_lamp/1phase_convection.i)
but to simply use water and to "convect" from the bottom. I'm working on this simple problem as a validation and learning exercise before I move onto more complex stuff.

I'm feeling confused because it seems to me that the heat isn't advecting through the timesteps.

I did some simple scaling analysis to see if it should look like this:

$$ t = \frac{L^{2}\rho c_{p}}{k} $$

where $t$ is the timescale for pure heat diffusion and $L$ is the size of the domain $rho$ is the mixture density, $c_{p}$ is mixture heat capacity, $k$ is the mixture thermal conductivity all as function of porosity.

The timescale for this domain size (2m) is $7x10e6$ s. After running the simulation for $10e7$ seconds, it doesn't appear to have changed at all. I did some other scaling of the porous flow Rayleigh number and it isn't high enough for there to be active convection but their still should be conduction I think.

![conduction_endtime1e7](https://github.com/idaholab/moose/assets/51497688/10d36fe8-20af-443d-b5dd-be2769b099b6)

The simulation seems to be running correctly and does converge appropriately.

I've been working on it for a while now and don't understand what is wrong.

I think it could be:
  1. Related to scaling
  2. Not using the right kernals
  3. Boundary conditions

I'm sure that this is some error that I've made in the input file. Thank you for your help.

For context, my goal is to work up to modeling a 5km domain around a dike intrusion at 800C with hydrothermal circulation and phase change. 

Here is the input file:
```


[Adaptivity]
  max_h_level = 2
  marker = marker
  initial_marker = initial
  initial_steps = 2
  [Indicators]
    [indicator]
      type = GradientJumpIndicator
      variable = temperature
    []
  []
  [Markers]
    [marker]
      type = ErrorFractionMarker
      indicator = indicator
      refine = 0.8
    []
    [initial]
      type = BoxMarker
      bottom_left = '0 1.95 0'
      top_right = '2 2 0'
      inside = REFINE
      outside = DO_NOTHING
    []
  []
[]

[Mesh]
  [gen]
    type = GeneratedMeshGenerator
    dim = 2
    ymin = 0
    ymax = 20
    xmax = 20
    ny = 100
    nx = 100
  []

  [dike]
    type = SubdomainBoundingBoxGenerator
    block_id = 1
    bottom_left = '0 0 0'
    top_right = '20 5 0'
    block_name = dike
    input = gen
  []
[]

[AuxVariables]
  [porosity]
    order = CONSTANT
    family = MONOMIAL
  []
[]

[GlobalParams]
  PorousFlowDictator = dictator
[]

[Variables]
  [porepressure]

  []
  [temperature]
    scaling = 1E-8
  []
[]

[PorousFlowBasicTHM]
  porepressure = porepressure
  temperature = temperature
  coupling_type = ThermoHydro
  gravity = '0 -9.81 0'
  fp = true_water
[]


[ICs]
  [porosity]
    type = RandomIC
    variable = porosity
    min = 0.25
    max = 0.350
    seed = 0
  []
  [pressure]
    type = ConstantIC
    variable = porepressure
    value = 10e6
  []
  [temperature1]
    type= ConstantIC
    variable = temperature
    value = 300
    block = 0
  []
  [temperature2]
    type= ConstantIC
    variable = temperature
    value = 372
    block = 'dike'
  []
[]

[BCs]
  [insulaing]
    type = NeumannBC
    value = 0
    variable = temperature
    boundary = 'right left top bottom'
  []
[]


[FluidProperties]
  [true_water]
    type = Water97FluidProperties
  []
[]

[Materials]
  [porosity]
    type = PorousFlowPorosity
    porosity_zero = 0.1
  []
  [biot_modulus]
    type = PorousFlowConstantBiotModulus
    biot_coefficient = 0.8
    solid_bulk_compliance = 2E-7
    fluid_bulk_modulus = 1E7
  []
  [permeability]
    type = PorousFlowPermeabilityConst
    permeability = '1E-14 0 0   0 1E-14 0   0 0 1E-14'
  []

  [thermal_expansion]
    type = PorousFlowConstantThermalExpansionCoefficient
    biot_coefficient = 0.8
    drained_coefficient = 0.003
    fluid_coefficient = 0.0002
  []
  [rock_internal_energy]
    type = PorousFlowMatrixInternalEnergy
    density = 2500.0
    specific_heat_capacity = 1200.
  []
  [thermal_conductivity]
    type = PorousFlowThermalConductivityIdeal
    dry_thermal_conductivity = '2.5 0 0  0 2.5 0  0 0 2.5'
  []
[]

[Preconditioning]
  active = mumps
  [mumps]
    type = SMP
    full = true
    petsc_options_iname = '-pc_type -pc_factor_mat_solver_package'
    petsc_options_value = ' lu       mumps'
  []
  [basic]
    type = SMP
    full = true
    petsc_options = '-ksp_diagonal_scale -ksp_diagonal_scale_fix'
    petsc_options_iname = '-pc_type -sub_pc_type -sub_pc_factor_shift_type -pc_asm_overlap'
    petsc_options_value = ' asm      lu           NONZERO      2'
  []
[]

[Executioner]
  type = Transient
  solve_type = NEWTON
  end_time = 10e7
  nl_max_its = 25
  l_max_its = 100
  dtmax = 1e4
  nl_abs_tol = 1e-4
  [TimeStepper]
    type = IterationAdaptiveDT
    dt = 100
  []
  line_search = 'none'
[]

[Outputs]
  print_linear_residuals = true
  perf_graph = true
  exodus = true
  csv = true
[]


```

**Comment by cpgr** [comment, marked as answer]
Your tolerance is too loose, so no solving takes place. When I run your input file, the output looks like
```
Time Step 0, time = 0

Time Step 1, time = 100, dt = 100
 0 Nonlinear |R| = 2.589940e-05
 Solve Converged!

Time Step 2, time = 300, dt = 200
 0 Nonlinear |R| = 2.589940e-05
 Solve Converged!

Time Step 3, time = 700, dt = 400
 0 Nonlinear |R| = 2.589940e-05
 Solve Converged!
```
In this case, the initial residual is less than your tolerance, so no solve takes place. Therefore, nothing evolves.

When I comment out the scaling for temperature and change nl_abs_tol to 1e-8, then the output looks like
```
Time Step 0, time = 0

Time Step 1, time = 100, dt = 100
 0 Nonlinear |R| = 2.588833e+03
      0 Linear |R| = 2.588833e+03
      1 Linear |R| = 1.482230e-11
 1 Nonlinear |R| = 2.693752e-01
      0 Linear |R| = 2.693752e-01
      1 Linear |R| = 2.806349e-15
 2 Nonlinear |R| = 6.235825e-09
 Solve Converged!
  Finished Solving         
```
In this case, something is happening!

**Reply by aikubo** [reply]
Thank you! I ran it for a short time and it does appear to be diffusing now.
